### PR TITLE
Add view support

### PR DESCRIPTION
**Pull request by ouyuanning**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3427 

## What this PR does / why we need it:
add view support in MO

### CODE REVIEW

**Review by fengttt — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Review by jianwan0214 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

File: pkg/vm/engine/memEngine/engine.go [hunk 1]
```diff
@@ -107,3 +107,7 @@ func (e *MemEngine) GetHideKeyDef(_ string, _ string) *plan.ColDef {
 func (e *MemEngine) Cost(_ *plan.ObjectRef, _ *plan.Expr) *plan.Cost {
 	return &plan.Cost{}
 }
+
+func (e *MemEngine) GetRootSql() string {
```
Comment (nnsgmsone) [inline pkg/vm/engine/memEngine/engine.go:111]: Why is there such a thing.

**Review by nnsgmsone — APPROVED** [review]

File: pkg/sql/plan/build_ddl.go [hunk 2]
```diff
@@ -15,15 +15,88 @@
...
+
+	viewData, err := json.Marshal(ViewData{
+		Stmt:            ctx.GetRootSql(),
+		DefaultDatabase: ctx.DefaultDatabase(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	createTable.TableDef.Defs = append(createTable.TableDef.Defs, &plan.TableDef_DefType{
```
Comment (nnsgmsone) [inline pkg/sql/plan/build_ddl.go:77]: please use tabledef instead of properties

Comment (nnsgmsone) [inline pkg/sql/plan/build_ddl.go:77]: please use tabledef instead of properties

**Review by nnsgmsone — CHANGES_REQUESTED** [review]

File: pkg/vm/engine/tae/moengine/helper.go [hunk 3]
```diff
@@ -64,6 +65,18 @@ func SchemaToDefs(schema *catalog.Schema) (defs []engine.TableDef, err error) {
 		}
 		defs = append(defs, pk)
 	}
+	pro := new(engine.PropertiesDef)
```
Comment (LeftHandCold) [inline pkg/vm/engine/tae/moengine/helper.go:68]: need to add judgment

**Review by XuPeng-SH — APPROVED** [review]
lgtm

File: pkg/vm/engine/tae/moengine/helper.go (hunk 3 above)
Comment (ouyuanning) [reply pkg/vm/engine/tae/moengine/helper.go]: thanks. fixed

File: pkg/sql/plan/build_ddl.go (hunk 2 above)
Comment (ouyuanning) [reply pkg/sql/plan/build_ddl.go:75]: get it.

**Review by nnsgmsone — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by LeftHandCold — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**